File: src/main/scala/com/teambrmodding/assistedprogression/managers/EventManager.scala
package com.teambrmodding.assistedprogression.managers

import com.teambrmodding.assistedprogression.common.items.ItemTrashBag
import com.teambrmodding.assistedprogression.events.RenderEvents
import net.minecraftforge.common.MinecraftForge

/**
  * This file was created for Assisted-Progression
  *
  * Assisted-Progression is licensed under the
  * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
  * http://creativecommons.org/licenses/by-nc-sa/4.0/
  *
  * @author Paul Davis - pauljoda
  * @since 1/18/2017
  */
object EventManager {

    /**
      * Adds all events we use to the event handler system
      */
    def registerEvents(): Unit = {
        MinecraftForge.EVENT_BUS.register(new RenderEvents)
        MinecraftForge.EVENT_BUS.register(ItemTrashBag)
    }
}

File: src/main/java/com/teambrmodding/assistedprogression/managers/EventManager.java
package com.teambrmodding.assistedprogression.managers;

import com.teambrmodding.assistedprogression.network.PacketManager;
import com.teambrmodding.assistedprogression.network.packet.ReadGrinderRecipePacket;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.entity.player.ServerPlayerEntity;
import net.minecraftforge.event.entity.player.PlayerEvent;
import net.minecraftforge.eventbus.api.EventPriority;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.network.PacketDistributor;

/**
 * This file was created for AssistedProgression
 * <p>
 * AssistedProgression is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis - pauljoda
 * @since 8/28/2019
 */
public class EventManager {

    @SuppressWarnings("unchecked")
    @SubscribeEvent(priority = EventPriority.HIGHEST)
    public static void onPlayerLogin(PlayerEvent.PlayerLoggedInEvent event) {
        PlayerEntity player = event.getPlayer();
        if(player instanceof ServerPlayerEntity)
            PacketManager.INSTANCE.send(PacketDistributor.PLAYER.with(() -> (ServerPlayerEntity) event.getPlayer()),
                    new ReadGrinderRecipePacket());
    }
}
